vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c

```diff
@@ -935,16 +935,36 @@ BTStatus_t prvBTSetAdvData( uint8_t ucAdapterIf,
     if( xErrCode == NRF_SUCCESS )
     {
         xAdvConfig.ble_adv_fast_enabled = true;
-        xAdvConfig.ble_adv_fast_interval = IOT_BLE_ADVERTISING_INTERVAL;
+        xAdvConfig.ble_adv_slow_enabled = true;
+
+        if( pxParams->usMinAdvInterval != 0 )
+        {
+            xAdvConfig.ble_adv_fast_interval = pxParams->usMinAdvInterval;
+        }
+        else
+        {
+            xAdvConfig.ble_adv_fast_interval = IOT_BLE_ADVERTISING_INTERVAL;
+        }
+
+        if( pxParams->usMaxAdvInterval != 0 )
+        {
+            xAdvConfig.ble_adv_slow_interval = pxParams->usMaxAdvInterval;
+        }
+        else
+        {
+            xAdvConfig.ble_adv_slow_interval = ( IOT_BLE_ADVERTISING_INTERVAL * 2 );
+        }
 
         /* If it is an advertisement data and advertising timeout is provided, set the timeout. Else set the default timeout by vendor. */
-        if( ( pxParams->bSetScanRsp == false ) && ( pxParams->usTimeout != 0 ) )
+        if( pxParams->usTimeout != 0 )
         {
             xAdvConfig.ble_adv_fast_timeout = pxParams->usTimeout;
+            xAdvConfig.ble_adv_slow_timeout = pxParams->usTimeout;
```

Comment:
There's already a block above that set these fields. Why are we doing it again here?

Reply:
Those are advertisement intervals and this is for advertising timeout. Previously we only had fast advertising mode for nordic but now since advertising interval ranges are supported we need both fast and slow adv modes.